src/main/java/com/strandls/cca/service/impl/CCADataServiceImpl.java

```diff
@@ -228,23 +233,38 @@ public AggregationResponse getCCADataList(HttpServletRequest request, UriInfo ur
 
 	@SuppressWarnings({ "unchecked", "rawtypes" })
 	@Override
-	public Map<String, Object> getCCADataAggregation(HttpServletRequest request, UriInfo uriInfo, boolean myListOnly)
+	public Map<String, Object> getCCADataAggregation(String query,HttpServletRequest request, UriInfo uriInfo, boolean myListOnly)
```

Comment:
make sure that relevant changes are made in it's controller function as well UI calling that endpoint and that it is not breaking existing code.